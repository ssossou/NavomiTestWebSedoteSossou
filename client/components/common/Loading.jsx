import React from 'react';
import PropTypes from 'prop-types';
import { Col, ProgressBar, Row } from 'react-materialize';

/**
 * displays a loading progress bar
 *
 * @param {Object} props object with property, text
 *
 * @returns {JSX} JSX element
 */
const Loading = props => (
  <section className="loading-wrapper center">
    <Row>
      <Col s={12}>
        <strong>{props.text}</strong>
        <ProgressBar />
      </Col>
    </Row>
  </section>
);

Loading.propTypes = {
  text: PropTypes.string,
};

export default Loading;
