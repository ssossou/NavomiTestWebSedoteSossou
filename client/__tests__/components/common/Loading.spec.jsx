import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Loading from '../../../components/common/Loading';


describe('Loading', () => {
  const wrapper = shallow(<Loading />);
  it('renders properly with defaults', () => {
    const LoadingComponent = renderer.create(<Loading />).toJSON();
    expect(LoadingComponent).toMatchSnapshot();
  });

  it('renders text prop passed to it', () => {
    const LoadingComponent = renderer
      .create(<Loading text="fetching"/>).toJSON();
    expect(LoadingComponent).toMatchSnapshot();
  });
  it('should render 1 ProgressBar component', () => {
    expect(wrapper.find('ProgressBar')).toHaveLength(1);
  });
  it('should render 1 Row component', () => {
    expect(wrapper.find('Row')).toHaveLength(1);
  });
  it('should render text prop within a strong tag', () => {
    const withProps = shallow(<Loading text="hello" />);
    expect(withProps.find('strong').text()).toEqual('hello');
  });
});
