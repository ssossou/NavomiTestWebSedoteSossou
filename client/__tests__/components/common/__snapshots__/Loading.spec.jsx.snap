// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders properly with defaults 1`] = `
<section
  className="loading-wrapper center"
>
  <div
    className="row"
  >
    <div
      className="col s12"
    >
      <strong />
      <div
        className="progress lighten-5"
      >
        <div
          className="indeterminate"
          style={undefined}
        />
      </div>
    </div>
  </div>
</section>
`;

exports[`Loading renders text prop passed to it 1`] = `
<section
  className="loading-wrapper center"
>
  <div
    className="row"
  >
    <div
      className="col s12"
    >
      <strong>
        fetching
      </strong>
      <div
        className="progress lighten-5"
      >
        <div
          className="indeterminate"
          style={undefined}
        />
      </div>
    </div>
  </div>
</section>
`;
